### ISSUE TITLE
zef install should tell user where the install was from.

### ISSUE THREAD

**Issue by coke** [ugexe/zef#261]
## Context
```
$zef install --force-install LWP::Simple
===> Searching for: LWP::Simple
!!!> Failed to update cpan mirror: https://raw.githubusercontent.com/ugexe/Perl6-ecosystems/master/cpan1.json
!!!> Failed to update cpan mirror: https://raw.githubusercontent.com/ugexe/Perl6-ecosystems/master/cpan.json
===> Updated cpan mirror: git://github.com/ugexe/Perl6-ecosystems.git
===> Updated p6c mirror: http://ecosystem-api.p6c.org/projects1.json
===> Testing: LWP::Simple:ver<0.104>:auth<github:perl6>
===> Testing [OK] for LWP::Simple:ver<0.104>:auth<github:perl6>
===> Installing: LWP::Simple:ver<0.104>:auth<github:perl6>

2 bin/ scripts [lwp-download.pl lwp-get.pl] installed to:
/Users/coke/.rakudobrew/moar-master/install/share/perl6/site/bin
```

## Expected Behavior
Output should include the source of the module that was installed. while the auth points us in a direction, we should have a URI for the source used, as well as (perhaps) a link to the bug tracker for that project (if defined) so that if there's a problem during the build/test we can report it to the correct location (very helpful if there are forks or multiple versions)

```
$ perl6 -v
This is Rakudo version 2018.06-216-g7a7e5e96f built on MoarVM version 2018.06-230-g0daa14c87
implementing Perl 6.c.
$ zef locate zef
===> From Distribution: zef:ver<0.4.5>:auth<github:ugexe>:api<>

===> From Distribution: zef:ver<0.4.4>:auth<github:ugexe>:api<>

===> From Distribution: zef:ver<0.4.3>:auth<github:ugexe>:api<>

===> From Distribution: zef:ver<0.4.2>:auth<github:ugexe>:api<>

===> From Distribution: zef:ver<0.2.9>:auth<github:ugexe>:api<>

===> From Distribution: zef:ver<0.2.3>:auth<github:ugexe>:api<>
```

**Comment by ugexe** [comment]
A full name can be used with `zef info` to get most of that, though source-url may have no meaning if it was recommended by `Zef::Repository::LocalCache`
```
$ zef info "LWP::Simple:ver<0.104>:auth<github:perl6>"
- Info for: LWP::Simple:ver<0.104>:auth<github:perl6>
- Identity: LWP::Simple:ver<0.104>:auth<github:perl6>
- Recommended By: Zef::Repository::Ecosystems<p6c>
- Installed: No
Author:	 Cosimo Streppone
Description:	 LWP::Simple quick & dirty implementation for Rakudo Perl 6
Source-url:	 https://github.com/perl6/perl6-lwp-simple.git
Provides: 1 modules
Depends: 3 items
```

**Comment by ugexe** [comment]
I've left a comment:
https://github.com/ugexe/zef/blob/53cbc71e411a0ce075ddd2a4644946f84a7601cf/lib/Zef/Client.pm6#L421-L426

Basically we would want to add this "downloaded from $url" message in `Zef::Client.fetch()`. But dists that are already local ( such as `zef install .` or `zef install $some-cached-dist` ) do not go through `fetch()` as a workaround to avoid `zef install .` from caching dev commits as a given name/api/ver/auth. This means that such a message would not show up when downloading from the local cache ( since the fetch phase is simply skipped ).

The workaround logic cannot just be moved into `fetch()`, because `zef fetch .` *should* cache a given distribution. Instead some sort of new logic will need to be put into `fetch()` that can identify which dists are appropriate to cache.

**Comment by ugexe** [comment]
This will be resolved in v0.8.0 with the following example output:
```
$ perl6 -I. bin/zef install HTTP::UserAgent --debug
===> Searching for: HTTP::UserAgent
===> Updating cpan mirror: https://raw.githubusercontent.com/ugexe/Perl6-ecosystems/master/cpan1.json
===> Updating p6c mirror: https://raw.githubusercontent.com/ugexe/Perl6-ecosystems/master/p6c1.json
===> Updated p6c mirror: https://raw.githubusercontent.com/ugexe/Perl6-ecosystems/master/p6c1.json
===> Updated cpan mirror: https://raw.githubusercontent.com/ugexe/Perl6-ecosystems/master/cpan1.json
===> Found: HTTP::UserAgent:ver<1.1.48>:auth<github:sergot> [via Zef::Repository::Ecosystems<p6c>]
===> Dependencies: HTTP::Status, File::Temp, DateTime::Parse, Encode, MIME::Base64, URI, IO::Socket::SSL, IO::Capture::Simple, Test::Util::ServerPort
===> Searching for missing dependencies: HTTP::Status, File::Temp, DateTime::Parse, Encode, MIME::Base64, URI, IO::Socket::SSL, IO::Capture::Simple, Test::Util::ServerPort
===> Found dependencies: MIME::Base64:ver<1.2.1>:auth<github:retupmoca>, DateTime::Parse:ver<0.9.1>, URI:ver<0.3.0>, File::Temp:ver<0.0.6>, Encode:ver<0.0.2>:auth<github:sergot>, IO::Socket::SSL:ver<0.0.1>:auth<github:sergot>, IO::Capture::Simple, HTTP::Status [via Zef::Repository::Ecosystems<p6c>]
===> Found dependencies: Test::Util::ServerPort:ver<0.0.2>:auth<github:jonathanstowe>:api<1.0> [via Zef::Repository::Ecosystems<cpan>]
===> Dependencies: Test, File::Directory::Tree, OpenSSL
===> Searching for missing dependencies: File::Directory::Tree, OpenSSL
===> Found dependencies: OpenSSL:ver<0.1.22>:auth<github:sergot>, File::Directory::Tree:auth<labster> [via Zef::Repository::Ecosystems<p6c>]
===> Fetching: HTTP::UserAgent
===> Fetching: MIME::Base64
===> Fetching: DateTime::Parse
===> Fetching: URI
===> Fetching: File::Temp
[HTTP::UserAgent] Fetching git://github.com/sergot/http-useragent.git with plugin: Zef::Service::Shell::git+{<anon|1>}
[MIME::Base64] Fetching git://github.com/perl6/Perl6-MIME-Base64.git with plugin: Zef::Service::Shell::git+{<anon|1>}
[File::Temp] Fetching git://github.com/perlpilot/p6-File-Temp.git with plugin: Zef::Service::Shell::git+{<anon|1>}
[DateTime::Parse] Fetching git://github.com/sergot/datetime-parse.git with plugin: Zef::Service::Shell::git+{<anon|1>}
[URI] Fetching git://github.com/perl6-community-modules/uri.git with plugin: Zef::Service::Shell::git+{<anon|1>}
===> Fetching [OK]: MIME::Base64:ver<1.2.1>:auth<github:retupmoca> to /Users/nicklogan/.zef/tmp/1571939800.70566.3339.4754909559456/Perl6-MIME-Base64.git
===> Fetching [OK]: File::Temp:ver<0.0.6> to /Users/nicklogan/.zef/tmp/1571939800.70566.5026.183940802509/p6-File-Temp.git
===> Fetching [OK]: DateTime::Parse:ver<0.9.1> to /Users/nicklogan/.zef/tmp/1571939800.70566.5700.697911145824/datetime-parse.git
===> Fetching [OK]: HTTP::UserAgent:ver<1.1.48>:auth<github:sergot> to /Users/nicklogan/.zef/tmp/1571939800.70566.7950.947982081263/http-useragent.git
===> Fetching: Test::Util::ServerPort

...
```

**Comment by ugexe** [comment]
This should be resolved in v0.8.0 when using `--debug`
